src/entities.rs

```diff
@@ -182,7 +188,7 @@ pub(crate) struct Entities {
     // and then from the new IDs, using only a single atomic subtract.
     //
     // Once `flush()` is done, `free_cursor` will equal `pending.len()`.
-    pending: Vec<u32>,
+    pub pending: Vec<u32>,
```

Comment:
Exposing this directly vastly complicates reasoning about the data structure's invariants. Instead, add methods that expose the required functionality without interfering with invariants.